pkg/cmd/task/describe.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const describeTemplate = `Name:	{{ .Task.Name }}
+Namespace:	{{ .Task.Namespace }}
+
+Input Resources
+{{- if not .Task.Spec.Inputs }}
+No input resources
+{{- else }}
+{{- if eq (len .Task.Spec.Inputs.Resources) 0 }}
+No input resources
+{{- else }}
+NAME	TYPE
+{{- range $ir := .Task.Spec.Inputs.Resources }}
+{{ $ir.Name }}	{{ $ir.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Output Resources
+{{- if not .Task.Spec.Outputs }}
+No output resources
+{{- else }}
+{{- if eq (len .Task.Spec.Outputs.Resources) 0 }}
+No output resources
+{{- else }}
+NAME	TYPE
+{{- range $or := .Task.Spec.Outputs.Resources }}
+{{ $or.Name }}	{{ $or.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Params
+{{- if not .Task.Spec.Inputs }}
+No params
+{{- else }}
+{{- if eq (len .Task.Spec.Inputs.Params) 0 }}
+No params
+{{- else }}
+NAME	TYPE
+{{- range $p := .Task.Spec.Inputs.Params }}
+{{ $p.Name }}	{{ $p.Type }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+Steps
+{{- if eq (len .Task.Spec.Steps) 0 }}
+No steps
+{{- else }}
+NAME
+{{- range $step := .Task.Spec.Steps }}
+{{ $step.Name }}
+{{- end }}
+{{- end }}
+
+Taskruns
+{{- if eq (len .TaskRuns.Items) 0 }}
+No taskruns
+{{- else }}
+NAME	STARTED	DURATION	STATUS
+{{ range $tr:=.TaskRuns.Items }}
+{{- $tr.Name }}	{{ formatAge $tr.Status.StartTime $.Time}}	{{ formatDuration $tr.Status.StartTime $tr.Status.CompletionTime }}	{{ formatCondition $tr.Status.Conditions }}
+{{ end }}
+{{- end }}
+`
+
+func describeCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("describe")
+	eg := `
+# Describe a task of name 'foo' in namespace 'bar'
+tkn task describe foo -n bar
+
+tkn t desc foo -n bar
+`
+
+	c := &cobra.Command{
+		Use:     "describe",
+		Aliases: []string{"desc"},
+		Short:   "Describes a task in a namespace",
+		Example: eg,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Args:         cobra.MinimumNArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+			return printTaskDescription(s, p, args[0])
+		},
+	}
+
+	_ = c.MarkZshCompPositionalArgumentCustom(1, "__tkn_get_task")
+	f.AddFlags(c)
+	return c
+}
+
+func printTaskDescription(s *cli.Stream, p cli.Params, tname string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return err
+	}
+
+	task, err := cs.Tekton.TektonV1alpha1().Tasks(p.Namespace()).Get(tname, metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintf(s.Err, "failed to get task %s\n", tname)
+		return err
+	}
+	opts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("tekton.dev/task=%s", tname),
+	}
+	taskRuns, err := cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).List(opts)
+	if err != nil {
+		fmt.Fprintf(s.Err, "failed to get taskruns for task %s \n", tname)
```

Comment:
Please test this scenario also.

Reply:
we don't have a `fake` TektonClient, isnt it ?

Reply:
A respective test has been added. Thanks